bundles/org.openhab.core.thing/src/test/java/org/openhab/core/thing/binding/BaseDynamicStateDescriptionProviderTest.java

```diff
@@ -0,0 +1,112 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.thing.binding;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.openhab.core.events.Event;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.events.ChannelDescriptionChangedEvent;
+import org.openhab.core.thing.events.ChannelDescriptionChangedEvent.CommonChannelDescriptionField;
+import org.openhab.core.thing.link.ItemChannelLinkRegistry;
+import org.openhab.core.types.StateOption;
+import org.osgi.framework.BundleContext;
+
+/**
+ * Tests for {@link BaseDynamicStateDescriptionProvider}.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.WARN)
+class BaseDynamicStateDescriptionProviderTest {
+
+    private static final ThingTypeUID THING_TYPE_UID = new ThingTypeUID("binding:type");
+    private static final ThingUID THING_UID = new ThingUID(THING_TYPE_UID, "id");
+    private static final ChannelUID CHANNEL_UID = new ChannelUID(THING_UID, "channel");
+
+    @Mock
+    EventPublisher mockEventPublisher;
+
+    @Mock
+    ItemChannelLinkRegistry mockItemChannelLinkRegistry;
+
+    class TestBaseDynamicStateDescriptionProvider extends BaseDynamicStateDescriptionProvider {
+
+        public TestBaseDynamicStateDescriptionProvider() {
+            this.bundleContext = mock(BundleContext.class);
+            this.eventPublisher = mockEventPublisher;
+            this.itemChannelLinkRegistry = mockItemChannelLinkRegistry;
+        }
+    };
+
+    private BaseDynamicStateDescriptionProvider subject;
+
+    @BeforeEach
+    public void setup() {
+        when(mockItemChannelLinkRegistry.getLinkedItemNames(CHANNEL_UID)).thenReturn(Set.of("item1", "item2"));
+
+        subject = new TestBaseDynamicStateDescriptionProvider();
+    }
+
+    @Test
+    public void setStatePatternPublishesEvent() {
+        subject.setStatePattern(CHANNEL_UID, "%s");
+
+        ArgumentCaptor<Event> capture = ArgumentCaptor.forClass(Event.class);
+        verify(mockEventPublisher, times(1)).post(capture.capture());
+
+        Event event = capture.getValue();
+        assertTrue(event instanceof ChannelDescriptionChangedEvent);
+        ChannelDescriptionChangedEvent cdce = (ChannelDescriptionChangedEvent) event;
+        assertEquals(CommonChannelDescriptionField.PATTERN, cdce.getField());
+
```

Comment:
Suggestion: you could additionally checked the linked items and the pattern in the event.